src/utils/random.cpp

```diff
@@ -78,6 +107,7 @@ bool save_rng_to_checkpoint_shared(persist& p, const lbann_comm* comm) {
   rng_name = dirname + "/rng_seq_generator";
```

Comment:
Apparently, we have never been saving this RNG's state correctly.

Reply:
Noted